src/componentChangeDetectionRule.ts

```diff
@@ -0,0 +1,113 @@
+import { Utils } from 'tslint/lib';
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+import { getDecoratorArgument, getDecoratorName } from './util/utils';
+import { sprintf } from 'sprintf-js';
+import { NgWalker } from './angular';
+
+const OPTION_DEFAULT = 'Default';
+const OPTION_ON_PUSH = 'OnPush';
+
+export type ChangeDetectionType = 'Default' | 'OnPush';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static metadata: Lint.IRuleMetadata = {
+    ruleName: 'component-change-detection',
+    type: 'functionality',
+    description: 'Enforce the preferred component change detection type (Default or OnPush).',
```

Comment:
I'm not sure it's necessary to have an option.  We want to have a check on use of `OnPush`, not of `Default`

Reply:
Thanks so much for the quick review! I was debating about having the option as well, I suppose if angular ever offered a 3rd change detection strategy then we could just add an option for that that default it to OnPush so it wasn't breaking. WDYT?

Reply:
In your implementation, if we use the `DEFAULT` option, we have to set the property `changeDetection: ChangeDetectionStrategy.Default`. It's weird for a default property, no ?

Reply:
Yeah agreed, that is kind of weird, I'll just remove the option and keep it as `OnOpen` 👍